mmcv/runner/hooks/checkpoint.py

```diff
@@ -60,6 +60,16 @@ def _save_checkpoint(self, runner):
             self.out_dir = runner.work_dir
         runner.save_checkpoint(
             self.out_dir, save_optimizer=self.save_optimizer, **self.args)
+        if runner.meta is not None:
+            if self.by_epoch:
+                cur_ckpt_filename = self.args.get(
+                    'filename_tmpl', 'epoch_{}.pth').format(runner.epoch + 1)
```

Comment:
Why `'filename_tmpl'` is passed using implicit `kwargs`?

Reply:
Arguments of `runner.save_checkpoint()` are hidden from CheckpointHook, and different runners have different default values.

Reply:
It will be used for `runner.save_checkpoint(self.out_dir, save_optimizer=self.save_optimizer, **self.args)` also.